### PR TITLE
Add tutorial for region arguments

### PR DESCRIPTION
**Pull request by willschlitzer**
As discussed in #153, this is a tutorial that goes over the different types of arguments that can be passed to `region`.

### CODE REVIEW

File: doc/index.rst [hunk 1]
```diff
@@ -32,6 +32,7 @@
     projections/index.rst
     tutorials/coastlines.rst
     tutorials/plot.rst
+    tutorials/region-arguments.rst
```
Comment (seisman) [inline doc/index.rst:35]: Two suggestions:

1. Rename the tutorial to `regions.rst`
2. Move it before `plot.rst`

```suggestion
    tutorials/coastlines.rst
    tutorials/regions.rst
    tutorials/plot.rst
```

File: examples/tutorials/region-arguments.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
+"""
+Set the region argument
+=======================
+
+Many of the plotting functions take an argument for the ``region`` argument, which sets
```
Comment (seisman) [inline examples/tutorials/region-arguments.py:5]:
```suggestion
Many of the plotting functions take the ``region`` argument, which sets
```

File: examples/tutorials/region-arguments.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    frame="ag",
+)
+fig.show()
+
+########################################################################################
+#
+# GMT has the option for region coordinates to be passed in classic mode, which was the
+# standard prior to GMT version 6. The string format takes the coordinates for the
+# bottom-left and top-right coordinates. To specify classic mode, append **+r** at the
+# end of the ``region`` string.
```
Comment (seisman) [inline examples/tutorials/region-arguments.py:61]: I don't think the descriptions are correct. 

The _xlleft/ylleft/xuright/yuright_**+r** syntax is usually used for oblique projections. It's not related to classic mode.

https://docs.generic-mapping-tools.org/latest/cookbook/options.html#data-domain-or-map-region-the-r-option

File: examples/tutorials/region-arguments.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    frame="ag",
+)
+fig.show()
+
+########################################################################################
+# ISO code
+# --------
+#
+# The ``region`` can be set to include a specific area specified by the two-character
+# `ISO 3166-1 alpha-2 convention
```
Comment (seisman) [inline examples/tutorials/region-arguments.py:122]:
```suggestion
# ISO 3166-1 alpha-2 convention
```

File: examples/tutorials/region-arguments.py [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    shorelines="1/0.5p",
+    frame="ag",
+)
+fig.show()
+
+########################################################################################
+#
+# The area encompassed by the ISO code can be expanded by appending **+r**\ *increment*
+# to the ISO code. The *increment* unit is in degrees and expands the range of the
+# region in all directions.
```
Comment (seisman) [inline examples/tutorials/region-arguments.py:142]: **+e** and **+R** also allowed, and *increment* can be different for 4 directions.

https://docs.generic-mapping-tools.org/latest/cookbook/options.html#data-domain-or-map-region-the-r-option

File: examples/tutorials/region-arguments.py (hunk 3 above)
Comment (willschlitzer) [reply examples/tutorials/region-arguments.py:61]: My mistake. Thought that was a holdover to the original GMT. I'll update it.

File: examples/tutorials/region-arguments.py (hunk 5 above)
Comment (willschlitzer) [reply examples/tutorials/region-arguments.py]: Added

File: examples/tutorials/region-arguments.py [hunk 6]
```diff
@@ -0,0 +1,235 @@
+"""
```
Comment (seisman) [inline examples/tutorials/region-arguments.py:1]: @willschlitzer You also need to rename this file to `regions.rst`

Comment (willschlitzer) [reply examples/tutorials/region-arguments.py]: Pushed.

File: examples/tutorials/regions.py [hunk 7]
```diff
@@ -0,0 +1,235 @@
+"""
+Set the region argument
+=======================
```
Comment (seisman) [inline examples/tutorials/regions.py:3]:
```suggestion
Set the region
==============
```

**Review by seisman — APPROVED** [review]

**Review by seisman — APPROVED** [review]
This PR looks good to me except for a small comment above.

I'll merge this PR two days later to give @GenericMappingTools/python-contributors some time for more review.

**Pull request merged**